### PR TITLE
Robot e2e updates

### PR DESCRIPTION
**Pull request by alexeigo**
updated robot e2e based on proposed automation starter kit
1) corrected display of user's first and last names: auth-passport.js and app.js
2) added new data-testid attribute where needed
3) added new structure for e2e/robot
3) added robot test cases for (based on fake user stories): login, navigator bar, counter
4) modified circle.yml to run smoke test or all enabled cases

### CODE REVIEW

**Comment by alexeigo** [comment]
@SethDavenport would you kindly review this pr when you have the time.

File: e2e/robot/exec/run_configurable.bash [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# Use > 0 to consume one or more arguments per pass in the loop (e.g.
+# some arguments don't have a corresponding value to go with it such
+# as in the --default example).
+# note: if this is set to > 0 the /etc/hosts part is not recognized ( may be a bug )
+echo "Processing arguments:"
+while [[ $# > 1 ]]
+do
+key="$1"
+
+case $key in
```
Comment (SethDavenport) [inline e2e/robot/exec/run_configurable.bash]: There's a command called getopts that helps with command line argument parsing in bash scripts, might be worth a look.

File: e2e/robot/exec/run_all_on_browserstack_device_ios.bash [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+include_tags=""
+exclude_tags="Disabled"
+test_variables=("BROWSER:IE" "LOGIN_URL:http://localhost:8080/")
+browserstack=()
+browserstack+=("BROWSERSTACK.ENABLED:True")
+browserstack+=("BROWSERSTACK.KEY:put_your_key_here")
+browserstack+=("BROWSERSTACK.USERNAME:put_your_username_here")
+browserstack+=("BROWSERSTACK.PLATFORM:MOBILE")
+browserstack+=("BROWSERSTACK.DEVICE:iPhone 6")
+browserstack+=("BROWSERSTACK.BRWOSER.NAME:ios")
```
Comment (SethDavenport) [inline e2e/robot/exec/run_all_on_browserstack_device_ios.bash]: Typo: brwoser

**Comment by alexeigo** [comment]
@SethDavenport please review. should be good to merge.

**Comment by SethDavenport** [comment]
:shipit:

**Pull request merged**